src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -409,8 +417,117 @@ public Object call() throws WorkflowException {
             throw new RuntimeException("cancelDVAsSystemUser() Unable to cancel DV: " + dvDocumentNumber, e);
         }
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean autoApproveDisbursementVouchersSpawnedByRecurringDvs() {
+        LOG.info("autoApproveDisbursementVouchersSpawnedByRecurringDvs: Entered.");
+        int approvalCount = 0;
+        int errorCount = 0;
+        Collection <String> dvDocIds = getRecurringDisbursementVoucherSearchDao().findSavedDvIdsSpawnedByRecurringDvForCurrentAndPastFiscalPeriods(getCurrentFiscalPeriodEndDate());
+        if (dvDocIds != null) {
+            for (String dvDocId :  dvDocIds) {
+                LOG.info("Processing start for DV ID:: " + dvDocId);
+                CuDisbursementVoucherDocument dv = null;
+
+                try{
+                    dv = (CuDisbursementVoucherDocument) getDocumentService().getByDocumentHeaderId(dvDocId);
+                } catch (WorkflowException e) {
+                    errorCount++;
+                    LOG.error("Could NOT find DV for Document ID::" + dvDocId);
+                    dv = null;
+                }
+
+                if (dv != null) {
+                    if (isKfsSystemUserDvInitiator(dv)) {
+                        try {
+                            addNoteToAutoApproveDv(dv, "Batch job Submit and Blanket Approve performed for DV spawned by Recurring DV.");
```

Comment:
Per Cathy
"Batch job Blanket Approve performed for DV spawned by Recurring DV."
is fine for now.